File: archives/C#/0492. Construct the Rectangle.cs
public class Solution {
    public int[] ConstructRectangle(int area) {
        int W=1;
        int L=area;
        for(int i=W+1;i*i<=area;i++){
            if (area%i==0 && Math.Abs(area/i-i)<Math.Abs(W-L)){
                W=i;
                L=area/i;
            }
        }
        return new int[]{L,W};
    }
}

File: archives/20190519python/0492. Construct the Rectangle.py
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        w,l=1,area
        lo,hi=w,l
        while lo<=area/lo:
            if lo<=area/lo and area%lo==0:
                w,l=area//lo,lo
            lo+=1
        return [w,l]
